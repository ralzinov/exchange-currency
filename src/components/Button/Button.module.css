.host {
    border: 0;
    color: #fff;
    padding: 12px 24px;
    font-size: 16px;
    line-height: 1;
    border-radius: 4px;
    position: relative;
    background: #2684FF;
    cursor: pointer;
}

.host:disabled {
    background: #b1b1b1;
    cursor: not-allowed;
}

.host:after {
    content: '';
    position: absolute;
    left: 0;
    top: 0;
    width: 100%;
    height: 100%;
    border-radius: 4px;
    transition: 0.2s ease opacity;
    will-change: opacity;
    background: #000;
    opacity: 0;
}


.host:not(:disabled):hover:after {
    opacity: 0.1;
}

.host:not(:disabled):active:after {
    opacity: 0.2;
}
