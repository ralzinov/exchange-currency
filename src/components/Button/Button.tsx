import * as React from 'react';
import classnames from 'classnames';
import styles from './Button.module.css';

export interface IButtonProps extends React.ButtonHTMLAttributes<unknown> {
    className?: string;
}

export const Button: React.FC<IButtonProps> = (props) => {
    return (
        <button {...props} className={classnames(styles.host, props.className)}>{props.children}</button>
    );
};
