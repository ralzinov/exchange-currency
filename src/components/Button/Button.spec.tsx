import React from 'react';
import ReactDOM from 'react-dom';
import {Button} from './Button';
import {create} from 'react-test-renderer';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Button />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('should render with props and children', () => {
    const clickSpy = jest.fn();
    const button = create(
        <Button className="test-class-name" onClick={clickSpy}>Label</Button>
    );
    expect(button.toJSON()).toMatchSnapshot();
});
