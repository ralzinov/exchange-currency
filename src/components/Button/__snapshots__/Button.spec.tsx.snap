// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with props and children 1`] = `
<button
  className="host test-class-name"
  onClick={[MockFunction]}
>
  Label
</button>
`;
